### PR TITLE
refactor show clause.

### PR DESCRIPTION
**Pull request by hanfei1991**
Use new plan logic to build show. Abandon filter operator.
@shenli @coocood @zimulala @tiancaiamao @XuHuaiyu PTAL

### CODE REVIEW

**Comment by zimulala** [comment]
Fix CI @hanfei1991

File: executor/builder.go [hunk 1]
```diff
@@ -64,9 +63,6 @@ func (b *executorBuilder) build(p plan.Plan) Executor {
 		return b.buildExecute(v)
 	case *plan.Explain:
 		return b.buildExplain(v)
-	case *plan.Filter:
```
Comment (shenli) [inline executor/builder.go]: Why ignore Filter?

File: plan/planbuilder.go [hunk 2]
```diff
@@ -361,37 +353,62 @@ func splitWhere(where ast.ExprNode) []ast.ExprNode {
...
 	if show.Pattern != nil {
-		conditions = append(conditions, show.Pattern)
+		expr, _, _, err := b.rewrite(show.Pattern, p, nil, false)
+		if err != nil {
+			b.err = errors.Trace(err)
+			return nil
+		}
+		conditions = append(conditions, expr)
 	}
 	if show.Where != nil {
```
Comment (XuHuaiyu) [inline plan/planbuilder.go]: why not use `buildSelection` here?

Comment (hanfei1991) [inline plan/planbuilder.go]: we should pass the ast.where to buildSelection. But in this case, we use both ast.PattenLikeExpr and ast.Where to constuct selection.

**Comment by shenli** [comment]
@XuHuaiyu PTAL

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**